opcua/common/events.py

```diff
@@ -162,16 +179,19 @@ def get_event_obj_from_type_node(node):
         class CustomEvent(parent_eventtype):
 
             def __init__(self):
-                super(CustomEvent, self).__init__(extended=True)
+                parent_eventtype.__init__(self)
                 self.EventType = node.nodeid
                 curr_node = node
 
                 while curr_node.nodeid.Identifier != parent_identifier:
                     for prop in curr_node.get_properties():
-                        setattr(self, prop.get_browse_name().Name, prop.get_value())
+                        name = prop.get_browse_name().Name
+                        val = prop.get_data_value()
+                        self.add_property(name, val.Value.Value, val.Value.VariantType)
                     parents = curr_node.get_referenced_nodes(refs=ua.ObjectIds.HasSubtype, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
-                    if len(parents) != 1: # Something went wrong
-                        return None
```

Comment:
Because to check if we get only one parent. It just to be sure that structure is set-up right.